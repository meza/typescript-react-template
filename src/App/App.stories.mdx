import { Meta, Story, Canvas } from "@storybook/addon-docs/blocks";
import { MemoryRouter } from 'react-router-dom';
import App from './App';

<Meta title="Template App" component={App}/>

# This is the app
<Canvas>
  <Story name="Main App">
    <MemoryRouter initialEntries={['/']}><App/></MemoryRouter>
  </Story>
</Canvas>

<Canvas>
  <Story name="404 Not Found">
      <MemoryRouter initialEntries={['/gibberish']}><App/></MemoryRouter>
    </Story>
</Canvas>
