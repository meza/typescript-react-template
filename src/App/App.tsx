import React from 'react';
import './app.i18n';
import { useTranslation } from 'react-i18next';
import { Route, Routes } from 'react-router-dom';

const App = () => {
  const [t] = useTranslation();

  return (
    <div>
      <Routes>
        <Route path={'/'} element={<h1>{t('hello-world')}</h1>}/>
        <Route path={'*'} element={<h1>{t('Page Does Not Exist')}</h1>}/>
      </Routes>
    </div>
  );
};

export default App;
