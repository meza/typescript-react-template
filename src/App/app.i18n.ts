import i18n from 'i18next';
import { initReactI18next } from 'react-i18next';
// English
import en from '../Assets/locales/en.json';

/**
 * @todo Implement i18next support features
 * @body Check out the tools [here](https://www.i18next.com/overview/plugins-and-utils)
 */

export enum Language {
  English = 'en'
}

const resources = {
  [Language.English]: {
    translation: en
  }
};

i18n
  .use(initReactI18next) // passes i18n down to react-i18next
  .init({
    resources: resources,
    lng: Language.English, // Change current language on the end of this file
    interpolation: {
      escapeValue: false // react already safes from xss
    }
  });

export default i18n;

export const getCurrentLanguage = (): string => {
  return i18n.language;
};

export type LanguageReturnType = string[];

export const getLanguagesList = (): LanguageReturnType[] => [
  ['English', Language.English]
];

export const changeCurrentLanguage = (language: Language): void => {
  i18n.changeLanguage(language);
};

export const existsLanguageByCode = (code: string): boolean => {
  const languages: LanguageReturnType[] = getLanguagesList();
  let exists = false;

  languages.forEach((lang) => {
    if (lang[0].toLowerCase() === code.toLowerCase()) {
      exists = true;
      return;
    }
  });

  return exists;
};
