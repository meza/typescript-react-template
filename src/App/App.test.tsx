import React from 'react';
import renderer from 'react-test-renderer';
import App from './App';
import { MemoryRouter } from 'react-router-dom';
import { chance } from 'jest-chance';

jest.mock('./app.i18n.ts');

describe('The app', () => {
  it('renders the hello world', () => {
    const component = renderer.create(<MemoryRouter initialEntries={['/']}><App/></MemoryRouter>);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders the 404', () => {
    const random = chance.word();
    const component = renderer.create(<MemoryRouter initialEntries={[`/${random}`]}><App/></MemoryRouter>);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
