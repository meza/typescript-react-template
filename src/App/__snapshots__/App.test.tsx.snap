// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The app renders the 404 1`] = `
<div>
  <h1>
    t(Page Does Not Exist)
  </h1>
</div>
`;

exports[`The app renders the hello world 1`] = `
<div>
  <h1>
    t(hello-world)
  </h1>
</div>
`;
